src/main/scala/net/snowflake/spark/snowflake/Utils.scala

```diff
@@ -724,4 +724,64 @@ object Utils {
     }
     // scalastyle:off println
   }
+
+  // Find the first un-quoted char '.', return -1 if not found
+  private def findFirstUnquotedDotIndex(data: String): Int = {
+    var quoteCount = 0
+    for (i <- 0 until data.length) {
+      if (data(i) == '"') {
+        quoteCount = quoteCount + 1
+      } else if (data(i) == '.') {
+        if (quoteCount % 2 == 0) {
+          return i
+        }
+      }
+    }
+    -1
+  }
+
+  // Split object name with unquoted dot.
+  private[snowflake] def splitNameByDot(name: String): Seq[String] = {
+    val result = new mutable.ArrayBuffer[String]
+    var data = name
+    var done = false
+    while (!done) {
+      val dotIndex = findFirstUnquotedDotIndex(data)
+      if (dotIndex == -1) {
+        result.append(data)
+        done = true
+      } else {
+        result.append(data.substring(0, dotIndex))
+        data = data.substring(dotIndex + 1)
+      }
+    }
+    result
+  }
+
+  // Adjust table name by adding database or schema name for table existence check.
```

Comment:
1. The users should input a valid object name.
2. If it is invalid, this function just return the original name. it will fail in the query.
3. Totally, there are 4 valid format for the object name. SC only adjust 2 of them because the other 2 are fully qualified names. It is not necessary to adjust them.